### ISSUE TITLE
Feature idea • Merging queries while being batched if using same query

### ISSUE THREAD

**Issue by jeremyputeaux** [apollographql/apollo-link#325]
## Problem
I have a repetitive `subscription` witch leads to a repetitive query (if I'm not mistaken, since I need the `QueryRef` to execute the subscription `QueryRef.subscribeToMore`).

## Finding solution
So I implemented `apollo-link-batch-http` in order to avoid 20 queries and it works great.
**But it's doing the same query with only uuid changing.**

I thought that `dataloader` could help me out but it did not since they are separated queries (even if they are done in one call).

That 's why I thought that maybe it could be something that `apollo-link-batch-http` could to itself (or maybe another package like `apollo-link-batch-merge-http`).

**It could merge queries that are using the same `query`.**

What do you think?

**Comment by evans** [comment]
Absolutely, you could use [`apollo-link-dedup`](https://www.apollographql.com/docs/link/links/dedup.html). It's documentation is a bit sparse, so if you don't understand something. Let's talk about documenting it!

**Comment by amakhrov** [comment]
If I understand the topicstarter correctly, `link-dedup` doesn't solve the problem. It dedups exactly same requests (operation name, query, variables). **In the topicstarter's case variables are different**.

What is needed to achieve the goal is to essentially rewrite the query (originally it took a single uuid, now it needs to take an array of uuids). And then analyze the returned collection and split it into individual requested results to be provided back to the subscribers.

I don't see an automatic way to do that. There is even no guarantee that it's possible for an arbitrary graphql schema.
